src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -412,6 +423,13 @@ public unsafe ConsoleKeyInfo ReadKey(out bool previouslyProcessed)
                 }
 
                 MapBufferToConsoleKey(out key, out ch, out isShift, out isAlt, out isCtrl);
+
+                // Replace the '\r' char with '\n' to match windows behaviour.
+                if (convertCrToLf && key == ConsoleKey.Enter && ch == '\r')
```

Comment:
Does this need the `key == ConsokeKey.Enter` condition?

Reply:
It does not, but I figured that this fact is dependent on the implementation of `MapBufferToConsoleKey`, so thought I'd leave it around.